### PR TITLE
Follow-up on classification score visualizers

### PR DESCRIPTION
**Pull request by rebeccabilbro**
This PR is an attempt to finish up the work that @abatula started in [her recently merged PR](https://github.com/DistrictDataLabs/yellowbrick/pull/407), which now successfully ensures that `ClassificationScoreVisualizer` returns `self.score_` after the `score()` method is called! 

Here's what I've currently managed to do:

 - updated [`ROCAUC`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier/rocauc.py#L201) `score` method to return `self.score_` instead of `self.estimator.score`.  _Note: ran into [this issue](https://github.com/DistrictDataLabs/yellowbrick/issues/252) when trying to add test coverage, so skipped that step for now; see below TODO_
 - updated [`ClassBalance`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier/class_balance.py#L92) `score` method to call `self.draw()` and return `self.score_` and added test coverage
 - updated  [`ClassPredictionError`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier/class_balance.py#L258) `score` method to call `self.draw()` and return `self.score_` and added test coverage
 - updated [`ConfusionMatrix`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier/confusion_matrix.py#L192) `score` method to call `self.draw()` and return `self.score_` and added test coverage

Still to do:
- [x] update  [`ClassificationReport`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier/classification_report.py#L141) `score` method to call `self.draw()` and return `self.score_` and add test coverage
- [x] figure out what to do about adding a `test_score_returns_score()` to the tests for `ROCAUC` (maybe just go ahead and add it  & mark it `@pytest.mark.skip` until we can come back to #252 ?) 

Question:
- there's now a good amount of duplication in the tests across the different classification score visualizers... possibly a signal that we should refactor; let me know if you think I should approach the   `test_score_returns_score()` tests differently

### CODE REVIEW

**Comment by rebeccabilbro** [comment]
@bbengfort - oops! Just saw that you added [an issue for this](https://github.com/DistrictDataLabs/yellowbrick/issues/502) - lmk how you want to proceed

**Comment by bbengfort** [comment]
@rebeccabilbro just reference that issue from this PR if this fixes it; I'll review in a second.

File: tests/test_classifier/test_rocauc.py [hunk 1]
```diff
@@ -109,6 +109,11 @@ def test_binary_rocauc(self):
 
         # Score the visualizer
         s = visualizer.score(X_test, y_test)
+
+        # Test that score method successfully returns a value between 0 and 1
+        assert 0 <= s <= 1
```
Comment (bbengfort) [inline tests/test_classifier/test_rocauc.py]: A good place to put this test! Is there a similar place for multiclass? Would that be required?

File: tests/test_classifier/test_rocauc.py [hunk 2]
```diff
@@ -109,6 +109,11 @@ def test_binary_rocauc(self):
...
+
+        # Test that the score method returns a value between 0 and 1
         self.assertAlmostEqual(s, 0.93230159261495249)
```
Comment (bbengfort) [inline tests/test_classifier/test_rocauc.py]: This probably can be removed in favor of the assertion you added.

File: tests/test_classifier/test_classification_report.py [hunk 3]
```diff
@@ -221,3 +221,14 @@ def test_invalid_support(self):
...
             ClassificationReport(LinearSVC(), support="foo")
+
+    def test_score_returns_score(self):
+        """
+        Test that ClassificationReport score() returns a score between 0 and 1
+        """
+        viz = ClassificationReport(LinearSVC(random_state=42))
+        
+        viz.fit(self.binary.X.train, self.binary.y.train)
+        s = viz.score(self.binary.X.test, self.binary.y.test)
```
Comment (bbengfort) [inline tests/test_classifier/test_classification_report.py]: Nice work using the dataset fixture that the class was using!

File: tests/test_classifier/test_class_prediction_error.py [hunk 4]
```diff
@@ -108,3 +110,26 @@ def test_class_type(self):
...
+        X = data[[
+            "temperature", "relative_humidity", "light", "C02", "humidity"
+        ]]
+
+        y = data['occupancy']
+
+        # Convert X to an ndarray
+        X = X.copy().view((float, len(X.dtype.names)))
+
+        X_train, X_test, y_train, y_test = tts(X, y, test_size=0.2, random_state=42)
```
Comment (bbengfort) [inline tests/test_classifier/test_class_prediction_error.py]: For this test, it might be simpler to use the [`X` and `y` variables created on L38](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/tests/test_classifier/test_class_prediction_error.py#L38) if you wanted - but this is perfectly fine. `ClassPredictionErrorTests` does need to have an integration test with a real data set + pandas added (though that's not for the PR, that's just a general comment).

File: yellowbrick/classifier/class_balance.py [hunk 5]
```diff
@@ -223,10 +245,11 @@ def score(self, X, y, **kwargs):
         Returns
         -------
 
-        ax : the axis with the plotted figure
+        score_ : float
+            Global accuracy score
         """
 
-        # We're replying on predict to raise NotFitted
+        # We're relying on predict to raise NotFitted
```
Comment (bbengfort) [inline yellowbrick/classifier/class_balance.py]: Good catch!

File: yellowbrick/classifier/classification_report.py [hunk 6]
```diff
@@ -129,16 +138,21 @@ def score(self, X, y=None, **kwargs):
         scores = list(scores)
         scores[-1] = scores[-1] / scores[-1].sum()
 
-        # Create a mapping composed of precision,recall, presion, and support
+        # Create a mapping composed of precision, recall, F1, and support
         # to their respective values
         scores = map(lambda s: dict(zip(self.classes_, s)), scores)
-        self.scores_ = dict(zip(SCORES_KEYS, scores))
+        self.scores_map_ = dict(zip(SCORES_KEYS, scores))
```
Comment (bbengfort) [inline yellowbrick/classifier/classification_report.py]: This is very minor, though I've probably given it more thought than I should have. I like the change in `ClassBalance` to store what used to be `scores_` into a `predictions_` table. Keeping the `score_` property more consistent as a single value metric throughout the `ClassificationScoreVisualizer` subclasses. 

However, here I would ask that we maintain this property as `self.scores_`. Although this might create a bit of confusion between `score_` and `scores_`, I believe it is important here to maintain backward compatibility with folks who were looking to fetch deeper metrics of this class (I know I have quite a bit of code that uses this attribute).

**Comment by bbengfort** [comment]
Looks like I accidentally deleted my comment replying to your question about redundancy in the tests - you're right, there is a lot of code duplication, but I don't think you should approach it differently. In the future, we should make this one of the checks in `test.checks`, e.g. for things like `fit()` returns self, etc. But I think you did well with the tests - adding checks to tests that already evaluated score, and using the datasets already available on the class. 

Just for future, it may be easier to use `sklearn.datasets.make_classification` and the fixtures in `conftest.py` that are already split into train/test splits - just to reduce some of the boilerplate of setting up datasets; but like I said in the code review comments - some of the tests needed to have the "real world integration" and pandas tests added anyway.

File: tests/test_classifier/test_rocauc.py (hunk 1 above)
Comment (rebeccabilbro) [reply tests/test_classifier/test_rocauc.py]: Done

File: tests/test_classifier/test_rocauc.py (hunk 2 above)
Comment (rebeccabilbro) [reply tests/test_classifier/test_rocauc.py]: Done

File: tests/test_classifier/test_class_prediction_error.py (hunk 4 above)
Comment (rebeccabilbro) [reply tests/test_classifier/test_class_prediction_error.py]: Done

File: yellowbrick/classifier/classification_report.py (hunk 6 above)
Comment (rebeccabilbro) [reply yellowbrick/classifier/classification_report.py]: Reverted

**Comment by bbengfort** [comment]
Ready for me to merge when tests pass?

**Comment by rebeccabilbro** [comment]
@bbengfort yes, thank you!

**Comment by bbengfort** [comment]
Closes #502
